demos/platforms/stm32f103c8.cpp

```diff
@@ -41,12 +47,44 @@ resource_list initialize_platform()
                                     .baud_rate = 115200,
                                   });
 
+  auto cpu_frequency = hal::stm32f1::frequency(hal::stm32f1::peripheral::cpu);
+  static hal::cortex_m::dwt_counter steady_clock(cpu_frequency);
   static hal::stm32f1::output_pin led('C', 13);
+  static hal::stm32f1::output_pin spi_chip_select('A', 4);
+  spi_chip_select.level(true);
+
+  hal::spi* spi = nullptr;
+
+  if constexpr (use_bit_bang_spi) {
+    static hal::stm32f1::output_pin sck('A', 5);
+    static hal::stm32f1::output_pin copi('A', 6);
+    static hal::stm32f1::input_pin cipo('A', 7);
+    static hal::soft::bit_bang_spi::pins bit_bang_spi_pins{ .sck = &sck,
+                                                            .copi = &copi,
+                                                            .cipo = &cipo };
+
+    static hal::spi::settings bit_bang_spi_settings{
+      .clock_rate = 250.0_kHz,
+      .clock_polarity = false,
+      .clock_phase = true,
+    };
+    static hal::soft::bit_bang_spi bit_bang_spi(
+      bit_bang_spi_pins, steady_clock, bit_bang_spi_settings);
+    spi = &bit_bang_spi;
+  } else {
+    static hal::stm32f1::spi spi1(hal::bus<1>,
+                                  {
+                                    .clock_rate = 250.0_kHz,
+                                    .clock_polarity = false,
+                                    .clock_phase = false,
+                                  });
+    spi = &spi1;
+  }
```

Comment:
```suggestion
    static hal::spi::settings spi_settings{
      .clock_rate = 250.0_kHz,
      .clock_polarity = false,
      .clock_phase = true,
    };
    static hal::soft::bit_bang_spi bit_bang_spi(
      bit_bang_spi_pins, steady_clock, spi_settings);
    spi = &bit_bang_spi;
  } else {
    static hal::stm32f1::spi spi1(hal::bus<1>, spi_settings);
    spi = &spi1;
  }
```

You are using the same settings for both so it makes sense to just pass them both in to each.